onnxruntime/contrib_ops/cpu/transformers/beam_search.h

```diff
@@ -27,7 +27,6 @@ class BeamSearch : public IControlFlowKernel {
       : IControlFlowKernel(info),
         encoder_feeds_fetches_manager_(nullptr),
         decoder_feeds_fetches_manager_(nullptr),
-        cuda_stream_(nullptr),
```

Comment:
> cuda_stream_(nullptr), [](http://example.com/codeflow?start=7&length=23)

The content of this file and others are under CPU here and yet, there is a lot of CUDA/device related code. Is the placement right?

Reply:
those transformer contribute ops typically itself is a subgraph, so just like control flow ops, those transformer ops provide a "framework" implementation that can be shared by both CPU and GPU, so the placement is correct. It doesn't contains cuda device code, but because the abstraction is not idea, you will see in the implementation it named many variables with "cuda" as a hint it actually manipulate some cuda handles. 

With this PR, at least for the cuda_stream part, we provide a framework Stream abstraction so the code here looks more neutron.